### DISCUSSION TITLE
Windows 11 - Trying to create a bat file to execute command on start-up

### DISCUSSION THREAD

**Discussion by abelaulau** [liquidctl/liquidctl#415] [Q&A]
Hi, 

I am a complete noob when it comes to coding and this question might be stupid. I managed to install liquidctl and have it working when I am manually typing each commands in on every start-up. My codes are simple start venv then run the liquidctl commands.

```
venv\Scripts\activate
liquidctl -m Kraken set logo color  off
liquidctl -m Kraken set ring color tai-chi 350017 ff2608
liquidctl -m Kraken set fan speed  20 30  30 50  35 80  40 85  50 100
liquidctl -m Kraken set pump speed 80
```

I have been trying to make this into a bat file and so it makes it easier whenever I need to run them but with little success. Would anyone be able to give me some guidance on how to create a script that can do this. Thanks in advance.

**Comment by jonasmalacofilho** [comment]
What errors are you seeing, exactly?

Anyway, I think one issue might be that your .bat doesn't necessarily execute in the correct directory, and that you need to `cd` into it before activating the venv or executing liquidctl.

**Comment by abelaulau** [comment]
I am not getting any errors. Its more like I run the script then the cmd just open a window for a split second and closes itself. And the codes inside won't get executed. 

I have added the `cd` line but still experiencing the same issue. This is currently all the codes inside the bat file.

```
cd C:\Users\myusername
venv\Scripts\activate
liquidctl -m Kraken set logo color  off
liquidctl -m Kraken set ring color tai-chi 350017 ff2608
liquidctl -m Kraken set fan speed  20 30  30 50  35 80  40 85  50 100
liquidctl -m Kraken set pump speed 80
```

I have no experience writing scripts. I guess my actual question is do I need to include any other lines/commands to have this bat working? Thanks.

**Reply by jonasmalacofilho** [reply]
IIRC, you can add a `pause` command at the end of the script to keep the window open, so you can see what's going on.

Also, is the `venv` directory _where you installed liquidctl_ really in `C:\Users\myusername`?

**Comment by daj999** [comment]
I just ran into this same problem.  Use "call liquidctl" on each line.  That should fix your problem.
